src/app/screens/PeerlyScreens/DetailsScreen/index.tsx

```diff
@@ -1,33 +1,99 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {View, Text, StyleSheet, Image} from 'react-native';
 import colors from '../../../constant/colors';
 import RatingBar from '../Components/RatingBar';
+import ObjectionModal from '../Components/ObjectionModal';
+import {usePostReward, usePostObjection} from './details.hooks';
+import CenteredModal from '../Components/Modal';
+import RewardSuccessIcon from '../../../../assets/peerly/svg/rewardSuccess.svg';
+import SuccessIcon from '../../../../assets/peerly/svg/Vector.svg';
+
+const AppreciationDetailsScreen = ({route}) => {
+  const {cardId, appriciationList} = route.params;
+  const cardDetails = appriciationList.find(item => item.id === cardId);
+  const [reward, setReward] = useState(0);
+  const [reason, setReason] = useState('');
+  const [isObjectionModalVisible, setObjectionModalVisible] = useState(false);
+
+  const {
+    mutate: postReward,
+    isLoading: isLoadingPostReward,
+    isSuccess: isSuccessPostReward,
+    reset: resetPostReward,
+  } = usePostReward();
+
+  const {
+    mutate: postObjection,
+    isLoading: isLoadingPostObjection,
+    isSuccess: isSuccessPostObjection,
+    reset: resetPostObjection,
+  } = usePostObjection();
+
+  useEffect(() => {
+    if (cardDetails) {
+      setReward(cardDetails.given_reward_point);
+    }
+  }, [cardDetails]);
+
+  useEffect(() => {
+    if (isSuccessPostObjection && isObjectionModalVisible) {
+      setObjectionModalVisible(false);
+    }
+  }, [isSuccessPostObjection, isObjectionModalVisible]);
+
+  const handleReward = (point: number) => {
+    const payload = {
+      params: {
+        id: cardDetails.id,
+      },
+      body: {point: point},
+    };
+    postReward(payload);
+  };
+
+  const handleObjectionReason = () => {
+    const payload = {
+      params: {
+        id: cardDetails.id,
+      },
+      body: {reporting_comment: reason},
+    };
+    postObjection(payload);
+  };
 
-const AppreciationScreen = () => {
   return (
     <View style={styles.container}>
       <View style={{alignItems: 'center'}}>
         <Image
-          source={require('../../../../assets/images/profile.png')}
+          source={
+            cardDetails?.receiver_image_url
+              ? {uri: cardDetails.receiver_image_url}
+              : require('../../../../assets/images/profile.png')
+          }
           style={styles.avatar}
         />
       </View>
       <View style={{alignItems: 'center', position: 'absolute', right: 80}}>
```

Comment:
Please avoid using inline styles. We should use style sheets whenever possible.

Reply:
Akshay was working on style part so he added temporary inline style. when the beautification will start we will remove all inline styles.